### ISSUE TITLE
Emit error for name mismatch when defining Enum using functional syntax

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7025]
The following should generate an error for consistency with `TypeVar` and other named entities created with a functional syntax.

```python
  WrongName = Enum('Color', 'RED GREEN BLUE')  # Type checker error
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.348, which was just published. It will also be included in a future release of pylance.
